libraries/AP_Landing/AP_Landing.cpp

```diff
@@ -207,6 +208,28 @@ AP_Landing::AP_Landing(AP_Mission &_mission, AP_AHRS &_ahrs, AP_TECS *_tecs_Cont
     AP_Param::setup_object_defaults(this, var_info);
 }
 
+/*
+  return a location alt in cm as AMSL
+  assumes loc frame is either AMSL or ABOVE_TERRAIN
+*/
+int32_t AP_Landing::loc_alt_AMSL_cm(const Location &loc) const
+{
+    int32_t alt_cm;
+    // try first with full conversion
+    if (loc.get_alt_cm(Location::AltFrame::ABSOLUTE, alt_cm)) {
+        return alt_cm;
+    }
+    if (loc.get_alt_frame() == Location::AltFrame::ABOVE_TERRAIN) {
+        // if we can't get true terrain then assume flat terrain
+        // around home
+        return loc.alt + ahrs.get_home().alt;
```

Comment:
Would be nice to not reference the private-in-the-future `.alt` field directly in new code.